### PR TITLE
rpc: testmempoolaccept for list of transactions

### PR DESCRIPTION
**Pull request by maflcko**
This extends the testmempoolaccept rpc to run on a vector of transactions. The idea is to keep the current mempool immutable and only keep track of the differences to the underlying mempool.

This is based on the transaction pool layer idea from #13804. Comment from there:

This implements a layer around an immutable tx pool. The layer can be seen as a temporary throw-away shell that provides the same interface as `CTxMemPool`. Its primary purpose right now is to be passed into ATMP while testing acceptance of several (potentially depending) transaction and then to be discarded.

One use case could be to determine if smart contracts that are set up with multiple txs would be accepted by current consensus and policy rules.

In the future it could be extended to support recursive wrapping of layers or a way to commit changes that happened in the layer to the underlying pool or layer. Furthermore, it could be extended to be revivable after changes to the underlying layer happened. (As opposed to be single-use)

### CODE REVIEW

**Comment by ryanofsky** [comment]
> One use case could be to determine if smart contracts that are set up with multiple txs would be accepted by current consensus and policy rules.

I think I basically understand this change, but not the use cases. I would love an ELI5 of the above `testmempoolaccept` use case. Specifically who might be calling the `testmempoolaccept` to build what kind of application.

Also, it would be useful to know if you think other applications of transaction pool layers could be more or equally useful as `testmempoolaccept`. IIRC these layers could also help implementing dandelion? Basically, I'd love to see some kind of post like #15605 or #14895 that explains what motivated these changes and how they can be built on and used.

File: src/txmempool.cpp [hunk 1]
```diff
@@ -598,6 +690,48 @@ static void CheckInputsAndUpdateCoins(const CTransaction& tx, CCoinsViewCache& m
...
+    for (const CTxMemPoolEntry* e : all_txs) {
+        const CTransaction& tx = e->GetTx();
+        const auto it = *GetIter(tx.GetHash());
+        setEntries setParentCheck;
+        for (const CTxIn& txin : tx.vin) {
+            if (exists(txin.prevout.hash)) {
+                setParentCheck.insert(*GetIter(txin.prevout.hash));
+            }
+            const CTransaction* c_tx = GetConflictTx(txin.prevout);
+            assert(c_tx = &tx);
```
Comment (practicalswift) [inline src/txmempool.cpp]: 👀

**Comment by MarcoFalke** [comment]
This doesn't compile/pass tests and a better solution is available in #20833, so please review that.
